### PR TITLE
feat: add variable to skip getting labels from docker hub

### PR DESCRIPTION
**Pull request by raphaelzoellner**
## Changes

A self-hosted experimental environment variable `RENOVATE_X_DISABLE_DOCKER_HUB_LABELS` which disables label requests to https://index.docker.io/.

## Context

This implements the proposed option from #29582.

This aims to optionally reduce the impact on the DockerHub rate limit further in scenarios where labels are not required (e.g. dry-run lookups).

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-experimental.md [hunk 1]
```diff
@@ -39,6 +39,10 @@ You can set the config file Renovate should read with the `RENOVATE_CONFIG_FILE`
 
 The process that runs Renovate must have the correct permissions to delete the config file.
 
+## `RENOVATE_X_DISABLE_DOCKER_HUB_LABELS`
```
Comment (viceice) [inline docs/usage/self-hosted-experimental.md:42]:
```suggestion
## `RENOVATE_X_DOCKER_HUB_DISABLE_LABEL_LOOKUP`
```
I prefer this naming to correlate with other options.

File: docs/usage/self-hosted-experimental.md [hunk 2]
```diff
@@ -39,6 +39,10 @@ You can set the config file Renovate should read with the `RENOVATE_CONFIG_FILE`
...
+
+If set to any value, Renovate will skip attempting to get release labels (e.g. gitRef, sourceUrl) from manifest annotations for `https://index.docker.io`.
```
Comment (viceice) [inline docs/usage/self-hosted-experimental.md:44]: add a more descriptive text about the side effects. eg no changelogs because of no source url.

**Review by viceice — COMMENTED** [review]
@rarkins I would like to defer until the recent refactor of flags has landed.

File: docs/usage/self-hosted-experimental.md (hunk 1 above)
Comment (raphaelzoellner) [reply docs/usage/self-hosted-experimental.md]: I've updated the variable name accordingly.

File: docs/usage/self-hosted-experimental.md (hunk 2 above)
Comment (raphaelzoellner) [reply docs/usage/self-hosted-experimental.md:44]: I've added additional notes to the docs about side effects setting this variable may result in.

**Review by viceice — APPROVED** [review]

**Pull request merged**